src/transport.cpp

```diff
@@ -369,12 +363,37 @@ namespace quicr {
         SendCtrlMsg(conn_ctx, buffer);
     }
 
+    void Transport::SendFetch(ConnectionContext& conn_ctx,
+                              uint64_t subscribe_id,
+                              const FullTrackName& tfn,
+                              messages::ObjectPriority priority,
+                              messages::GroupOrder group_order,
+                              messages::GroupId start_group,
+                              messages::GroupId end_group)
+    {
+        MoqFetch fetch;
+        fetch.subscribe_id = subscribe_id;
+        fetch.track_namespace = tfn.name_space;
+        fetch.track_name.assign(tfn.name.begin(), tfn.name.end());
+        fetch.priority = priority;
+        fetch.group_order = group_order;
+        fetch.start_group = start_group;
+        fetch.start_object = 0; // Request the start of the group
+        fetch.end_group = end_group;
+        fetch.end_object = 0; // Request the whole group
+
+        Bytes buffer;
+        buffer.reserve(sizeof(MoqFetch) + tfn.name.size() + tfn.name_space.size());
+        buffer << fetch;
+
+        SendCtrlMsg(conn_ctx, buffer);
+    }
+
     void Transport::SendFetchError(ConnectionContext& conn_ctx,
                                    [[maybe_unused]] uint64_t subscribe_id,
                                    FetchError error,
                                    const std::string& reason)
     {
-
         auto fetch_err = MoqFetchError{};
         fetch_err.subscribe_id = 0x1;
         fetch_err.err_code = static_cast<uint64_t>(error);
```

Comment:
Not part of our PR, but this should have had a metric as well for fetch failures.